test/models/architecture_test.rb

```diff
@@ -74,4 +75,29 @@ class ArchitectureTest < ActiveSupport::TestCase
     architecture = Architecture.new :name => "../../etc/shadow"
     assert_equal "etc-shadow", architecture.bootfilename_efi
   end
+
+  test "update with multi names" do
+    architecture = FactoryBot.create(:architecture)
+    valid_name_list.each do |new_name|
+      architecture.name =  new_name
+      assert_valid architecture
+      assert_equal architecture.name, new_name
+    end
+  end
+
+  test "should not update with invalid names" do
+    architecture = Architecture.first
+    invalid_name_list.each do |name|
+      architecture.name = name
+      refute_valid architecture
+      assert_includes architecture.errors.keys, :name
+    end
+  end
+
+  test "should destroy architecture" do
```

Comment:
I wonder what this test is actually testing? the deleting active record model is a standard rails operation, we only verify that `.delete` method works. The thing that could be tested is this https://github.com/theforeman/foreman/blob/develop/app/models/architecture.rb#L7 which ensures it can't be deleted if there's at least one host or hostgroup assigned to this architecture, but this architecture is not assigned to anything. But this is already being tested on [line 18](https://github.com/ldjebran/foreman/blob/10f5b9354f7328f8efb0b462e50d9752678dfacb/test/models/architecture_test.rb#L18)

Reply:
There could be a bug in that logic that prevents deletion in general.

Reply:
because of that callback? ok, we can keep it here, it doesn't take much time I guess, @ldjebran please ignore this commit